tests/SimplifyTest.elm

```diff
@@ -10424,4 +10424,27 @@ a = (let b = c in (f x).e).f
 a = (let b = c in ((f x).e).f)
 """
                         ]
+        , test "should simplify record accesses for if/then/else expressions" <|
+            \() ->
+                """module A exposing (..)
+a = (if x then { f = 3 } else { z | f = 3 }).f
+"""
+                    |> Review.Test.run (rule defaults)
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Field access can be simplified"
+                            , details = [ "Accessing the field outside an if/then/else expression can be simplified to access the field inside it" ]
+                            , under = "(if x then { f = 3 } else { z | f = 3 }).f"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = (if x then ({ f = 3 }).f else ({ z | f = 3 }).f)
```

Comment:
Let's make an improvement in a future PR (I can do it if you want, but feel free to work on it) to not add parentheses unnecessarily. That will make for better diffs and less unnecessary fixes.

Reply:
I'd prefer it if you worked on it, to be honest

Reply:
That's fine, I'll do that at some point 👍

Reply:
I've actually achieved most of the no-redundant-parens result ^_^